Update laboratory1.sql

dimon19991 opened the pull request.

dimon19991 committed:
Update laboratory1.sql

MikeKurshakov commented on km63/Koltsov_Dmytro/laboratory1.sql:
```diff
@@ -1,2 +1,137 @@
+--Код відповідь:
+
+CREATE USER KOLTSOV IDENTIFIED BY studentpma;
+DEFAULT TABLESPACE "Dima";
```
DEFAULT TABLESPACE "USERS"; -- название TABLESPACE  неправильное, если оно не было создано перед этим шагом.

MikeKurshakov commented on km63/Koltsov_Dmytro/laboratory1.sql:
```diff
@@ -1,2 +1,137 @@
+
+ALTER USER KOLTSOV QUOTA 100M on USERS;
+
+GRANT "CONECT" to KOLTSOV;
```
GRANT "CONNECT" TO KOLTSOV ;-- пропущена буква N.

MikeKurshakov commented on km63/Koltsov_Dmytro/laboratory1.sql:
```diff
@@ -1,2 +1,137 @@
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
```
GRANT INSERT ANY TABLE TO KOLTSOV ;
GRANT ALTER ANY TABLE TO KOLTSOV ;
GRANT SELECT ANY TABLE TO KOLTSOV ;

MikeKurshakov commented on km63/Koltsov_Dmytro/laboratory1.sql:
```diff
@@ -1,2 +1,137 @@
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
```
Это не алгебра Кодда.
Надо узнать минимальную стоимость из таблицы, а не задавать константой.

Алгебра Кодда:
PROJECT CUSTOMERS TIMES (ORDERS TIMES ORDERITEMS)
(WHERE (CUSTOMERS.CUST_ID = ORDERS.CUST_ID)
AND(ORDERS.ORDER_NUM=ORDERITEMS.ORDER_NUM)
AND(ORDERITEMS.ITEM_PRICE> PROJECT ORDERITEMS {MIN(ORDERITEMS.ITEM_PRICE)}))
{CUSTOMERS.CUST_NAME};

SQL:
SELECT DISTINCT(CUSTOMERS.CUST_NAME)
FROM CUSTOMERS , ORDERS , ORDERITEMS 
WHERE
    CUSTOMERS.CUST_ID = ORDERS.CUST_ID
    AND
    ORDERS.ORDER_NUM=ORDERITEMS.ORDER_NUM
    AND 
    ORDERITEMS.ITEM_PRICE>(SELECT MIN(ORDERITEMS.ITEM_PRICE)
FROM ORDERITEMS);

MikeKurshakov commented on km63/Koltsov_Dmytro/laboratory1.sql:
```diff
@@ -1,2 +1,137 @@
+
+--Код відповідь:
+
+
```
Алгебра Кодда:
PROJECT CUSTOMERS TIMES ORDERS RENAME(CUST_ID, NEW_CUST_ID)
(WHERE (CUSTOMERS.CUST_ID = ORDERS.CUST_ID)
AND (CUSTOMERS.CUST_EMAIL IS NULL)) {CUSTOMERS.CUST_NAME};

SQL:
SELECT CUSTOMERS.CUST_NAME 
FROM CUSTOMERS,ORDERS
WHERE CUSTOMERS.CUST_ID = ORDERS.CUST_ID
AND CUSTOMERS.CUST_EMAIL IS NULL;

MikeKurshakov commented on km63/Koltsov_Dmytro/laboratory1.sql:
```diff
@@ -1,2 +1,137 @@
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
```
Алгебра Кодда:
RENAME(PROJECT VENDORS TIMES PRODUCTS RENAME(VEND_ID, NEW_VEND_ID)
(WHERE (VENDORS.VEND_ID = PRODUCTS.VEND_ID)
AND (LENGTH(Products.PROD_DESC) = PROJECT PRODUCTS
{MAX(length(PRODUCTS.PROD_DESC)}
)) {UPPER(VENDORS.VEND_NAME)},"vendor_name");

SQL:
SELECT UPPER(VENDORS.VEND_NAME) AS "vendor_name"
FROM VENDORS,PRODUCTS
WHERE VENDORS.VEND_ID=PRODUCTS.VEND_ID
AND LENGTH(Products.PROD_DESC) = (SELECT MAX(length(PRODUCTS.PROD_DESC))
FROM PRODUCTS)
;

MikeKurshakov commented on km63/Koltsov_Dmytro/laboratory1.sql:
```diff
@@ -1,2 +1,137 @@
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
```
Должно быть три сущности Сomputer и OS Windows и Сonnection(вспомогательная). Связь будет многие-ко-многим так, как на одном компьютере может быть не одна версия OS Windows и OS Windows стоят на многих компьютерах.

OS Windows:

-version - primary key, уникальная версия системы формата 000.000000.000, цифры от 0 до 9. 
(пример: 000.000000.000)
Код:^[0-9]{3}[.]{1}[0-9]{6}[.]{1}[0-9]{3}$
-description-описание системы, больше одного слова, ограничено 100 символами, точки, скобки, тире и нижнее подчеркивание.
(пример: A stable system for a 64-bit computer.)
Код: ^[A-z_0-9- ().]{1,100}$
-language-язык системы, строка из 20 литералов, первый заглавный, остальные строчные.(пример:English)
Код:^[A-Z]{1}[a-z]{19}$
-syste_bit_rate-розрядность системы, число от 0 до 99, дефис, слово bit. 
(пример: 64 - bit)
Код:^[0-9]{2} [-]{1} bit$

Сomputer:

-mac_address-primary key, mac address сетевой карты компьютера(уникальный для каждого компьютера), формат: числовая или буквенная комбинация через каждые два символа разделённая двоеточием.
(пример:01:23:45:67:0b:cd)
Код:^[a-fA-F0-9:]{17}$
-logo - логотип производителя, строка из 10 литералов, первый заглавный, остальные строчные.
(пример:Asus)
Код:^[A-Z]{1}[a-z]{9}$
-CPU - процессор, больше одного слова, ограничено 20 символами, точки, скобки, тире и нижнее подчеркивание.
(пример: Intel Pentium Dual-Core)
Код: ^[A-z_0-9- ().]{1,20}$
-model - модель компьютера, число от 0 до 99 дефис, число 00000 до 99999 дефис, число от 000000 до 999999
(пример: 12-34567-890123)
Код:^[0-9]{2}-[0-9]{5}-[0-9]{6}$

Сonnection(для связывания компьютера и системы):

-version_fk-foreign key для поля version в OS Windows, уникальная версия системы формата 000.000000.000, цифры от 0 до 9. 
(пример: 000.000000.000)
Код:^[0-9]{3}[.]{1}[0-9]{6}[.]{1}[0-9]{3}$
-mac_address_fk-foreign key для поля mac_address в Сomputer, mac address сетевой карты компьютера(уникальный для каждого компьютера), формат: числовая или буквенная комбинация через каждые два символа разделённая двоеточием.
(пример:01:23:45:67:0b:cd)
Код:^[a-fA-F0-9:]{17}$

dimon19991 committed:
Update laboratory2.sql

dimon19991 commented:
create table OS_WINDOWS 
(
   VERSION              VARCHAR2(14)         not null,
   DESCRIPTION          VARCHAR2(100)        not null,
   LANGUAGE             VARCHAR2(20)         not null,
   SYSTE_BIT_RATE       VARCHAR2(6)          not null,
   constraint PK_OS_WINDOWS primary key (VERSION)
);


create table SOFTWARE 
(
   Name                 VARCHAR2(50)         not null,
   data                 DATE                 not null,
   memory               NUMBER(10)           not null,
   constraint PK_SOFTWARE primary key (Name, data)
);


create table soft_is_inctall 
(
   Name                 VARCHAR2(50)         not null,
   data                 DATE                 not null,
   VERSION              VARCHAR2(14)         not null,
   constraint PK_SOFT_IS_INCTALL primary key (Name, data, VERSION)
);

alter table soft_is_inctall
   add constraint FK_SOFT_IS__SOFT_IS_I_SOFTWARE foreign key (Name, data)
      references SOFTWARE (Name, data);

alter table soft_is_inctall
   add constraint FK_SOFT_IS__SOFT_IS_I_OS_WINDO foreign key (VERSION)
      references OS_WINDOWS (VERSION);
      
      
create table СOMPUTER 
(
   mac_address          VARCHAR2(17)         not null,
   logo                 VARCHAR2(10)         not null,
   CPU                  VARCHAR2(20)         not null,
   model                NUMBER(6)            not null,
   constraint PK_СOMPUTER primary key (mac_address)
);

create table os_is_instull 
(
   mac_address          VARCHAR2(17)         not null,
   VERSION              VARCHAR2(14)         not null,
   constraint PK_OS_IS_INSTULL primary key (mac_address, VERSION)
);


alter table os_is_instull
   add constraint FK_COMP_HAS_OS foreign key (mac_address)
      references СOMPUTER (mac_address);

alter table os_is_instull
   add constraint FK_OS_IS_INST foreign key (VERSION)
      references OS_WINDOWS (VERSION);
![conceptual](https://user-images.githubusercontent.com/22871731/38957783-21d9f262-4364-11e8-80c3-7fb2e5d1db78.PNG)
![logical](https://user-images.githubusercontent.com/22871731/38957785-21fe6fa2-4364-11e8-8377-93e88eb42e0b.PNG)
![physical](https://user-images.githubusercontent.com/22871731/38957786-221a9d9e-4364-11e8-95f9-2644c2553661.PNG)


3а
PROJECT (ORDERS TIMES ORDERITEMS)
(WHERE (CUSTOMERS.CUST_ID = ORDERS.CUST_ID)
AND(ORDERS.ORDER_NUM=ORDERITEMS.ORDER_NUM)
AND(ORDERITEMS.ITEM_PRICE> PROJECT ORDERITEMS {MIN(ORDERITEMS.ITEM_PRICE)}))
{CUSTOMERS.CUST_NAME};

3b
SELECT UPPER(VENDORS.VEND_NAME) AS "vendor_name"
FROM VENDORS,PRODUCTS
WHERE VENDORS.VEND_ID=PRODUCTS.VEND_ID
AND LENGTH(Products.PROD_DESC) = (SELECT MAX(length(PRODUCTS.PROD_DESC))
FROM PRODUCTS);

3c
SELECT UPPER(VENDORS.VEND_NAME) AS "vendor_name"
FROM VENDORS,PRODUCTS
WHERE VENDORS.VEND_ID=PRODUCTS.VEND_ID
AND LENGTH(Products.PROD_DESC) = (SELECT MAX(length(PRODUCTS.PROD_DESC))
FROM PRODUCTS);

dimon19991 committed:
Update laboratory3.sql

dimon19991 commented:
CREATE TABLE "DIMA"."OS_IS_INSTULL" 
   (	"MAC_ADDRESS" VARCHAR2(17 BYTE) NOT NULL ENABLE, 
	"VERSION" VARCHAR2(14 BYTE) NOT NULL ENABLE, 
	 CONSTRAINT "PK_OS_IS_INSTULL" PRIMARY KEY ("MAC_ADDRESS", "VERSION")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS 
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM"  ENABLE, 
	 CONSTRAINT "FK_COMP_HAS_OS" FOREIGN KEY ("MAC_ADDRESS")
	  REFERENCES "DIMA"."СOMPUTER" ("MAC_ADDRESS") ENABLE, 
	 CONSTRAINT "FK_OS_IS_INST" FOREIGN KEY ("VERSION")
	  REFERENCES "DIMA"."OS_WINDOWS" ("VERSION") ENABLE
   ) SEGMENT CREATION IMMEDIATE 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 NOCOMPRESS LOGGING
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM" ;

  
  
    CREATE TABLE "DIMA"."OS_WINDOWS" 
   (	"VERSION" VARCHAR2(14 BYTE) NOT NULL ENABLE, 
	"DESCRIPTION" VARCHAR2(100 BYTE) NOT NULL ENABLE, 
	"LANGUAGE" VARCHAR2(20 BYTE) NOT NULL ENABLE, 
	"SYSTE_BIT_RATE" VARCHAR2(6 BYTE) NOT NULL ENABLE, 
	 CONSTRAINT "PK_OS_WINDOWS" PRIMARY KEY ("VERSION")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS 
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM"  ENABLE
   ) SEGMENT CREATION IMMEDIATE 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 NOCOMPRESS LOGGING
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM" ;
  
  
  
  CREATE TABLE "DIMA"."SOFT_IS_INCTALL" 
   (	"NAME" VARCHAR2(50 BYTE) NOT NULL ENABLE, 
	"DATA" DATE NOT NULL ENABLE, 
	"VERSION" VARCHAR2(14 BYTE) NOT NULL ENABLE, 
	 CONSTRAINT "PK_SOFT_IS_INCTALL" PRIMARY KEY ("NAME", "DATA", "VERSION")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS 
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM"  ENABLE, 
	 CONSTRAINT "FK_SOFT_SOFTWARE" FOREIGN KEY ("NAME", "DATA")
	  REFERENCES "DIMA"."SOFTWARE" ("NAME", "DATA") ENABLE, 
	 CONSTRAINT "FK_SOFT_WINDO" FOREIGN KEY ("VERSION")
	  REFERENCES "DIMA"."OS_WINDOWS" ("VERSION") ENABLE
   ) SEGMENT CREATION IMMEDIATE 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 NOCOMPRESS LOGGING
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM" ;
  
  
    CREATE TABLE "DIMA"."SOFTWARE" 
   (	"NAME" VARCHAR2(50 BYTE) NOT NULL ENABLE, 
	"DATA" DATE NOT NULL ENABLE, 
	"MEMORY" NUMBER(10,2) NOT NULL ENABLE, 
	 CONSTRAINT "PK_SOFTWARE" PRIMARY KEY ("NAME", "DATA")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS 
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM"  ENABLE
   ) SEGMENT CREATION IMMEDIATE 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 NOCOMPRESS LOGGING
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM" ;

  
  
  CREATE TABLE "DIMA"."СOMPUTER" 
   (	"MAC_ADDRESS" VARCHAR2(17 BYTE) NOT NULL ENABLE, 
	"LOGO" VARCHAR2(10 BYTE) NOT NULL ENABLE, 
	"CPU" VARCHAR2(20 BYTE) NOT NULL ENABLE, 
	"MODEL" NUMBER(6,0) NOT NULL ENABLE, 
	 CONSTRAINT "PK_СOMPUTER" PRIMARY KEY ("MAC_ADDRESS")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS 
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM"  ENABLE
   ) SEGMENT CREATION IMMEDIATE 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 NOCOMPRESS LOGGING
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM" ;

  INSERT INTO "DIMA"."СOMPUTER" (MAC_ADDRESS, LOGO, CPU, MODEL) VALUES ('04:7D:7B:97:0C:9F', 'ASUS', 'INTEL I7', '1001')
INSERT INTO "DIMA"."СOMPUTER" (MAC_ADDRESS, LOGO, CPU, MODEL) VALUES ('08:ED:B9:49:B2:E5', 'DELL', 'INTEL I5', '1002')
INSERT INTO "DIMA"."СOMPUTER" (MAC_ADDRESS, LOGO, CPU, MODEL) VALUES ('01:0C:CD:00:00:01', 'SUMSUNG', 'INTEL I3', '1003')
INSERT INTO "DIMA"."СOMPUTER" (MAC_ADDRESS, LOGO, CPU, MODEL) VALUES ('01:0C:CD:01:01:21', 'DELL', 'INTEL I7', '1004')


	INSERT INTO "DIMA"."OS_WINDOWS" (VERSION, DESCRIPTION, LANGUAGE, SYSTE_BIT_RATE) VALUES ('0.0.1', 'GOOD', 'English', '64-bit')
INSERT INTO "DIMA"."OS_WINDOWS" (VERSION, DESCRIPTION, LANGUAGE, SYSTE_BIT_RATE) VALUES ('1.15.2', 'BUG', 'Freanch', '32-bit')
INSERT INTO "DIMA"."OS_WINDOWS" (VERSION, DESCRIPTION, LANGUAGE, SYSTE_BIT_RATE) VALUES ('5.102.84', 'SLOW', 'Englich', '32-bit')
INSERT INTO "DIMA"."OS_WINDOWS" (VERSION, DESCRIPTION, LANGUAGE, SYSTE_BIT_RATE) VALUES ('12.15.2', 'GOOD', 'German', '64-bit')


	INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('04:7D:7B:97:0C:9F', '0.0.1')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('08:ED:B9:49:B2:E5', '5.102.84')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('04:7D:7B:97:0C:9f', '12.15.2')



INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('04:7D:7B:97:0C:9F', '0.0.1')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('08:ED:B9:49:B2:E5', '5.102.84')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('04:7D:7B:97:0C:9F', '12.15.2')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('01:0C:CD:00:00:01', '0.0.1')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('01:0C:CD:00:00:01', '1.15.2')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('01:0C:CD:01:01:21', '12.15.2')


INSERT INTO "DIMA"."SOFTWARE" (NAME, DATA, MEMORY) VALUES ('SKYPE', TO_DATE('2017-12-12 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '705,2')
INSERT INTO "DIMA"."SOFTWARE" (NAME, DATA, MEMORY) VALUES ('Visual Studio', TO_DATE('1017-09-01 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '2028.15')


INSERT INTO "DIMA"."SOFT_IS_INCTALL" (NAME, DATA, VERSION) VALUES ('Skype', TO_DATE('2017-12-12 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '0.0.1')
INSERT INTO "DIMA"."SOFT_IS_INCTALL" (NAME, DATA, VERSION) VALUES ('Team Viver', TO_DATE('2018-10-15 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '12.15.2')
INSERT INTO "DIMA"."SOFT_IS_INCTALL" (NAME, DATA, VERSION) VALUES ('Viber', TO_DATE('2012-10-10 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '1.15.2')
INSERT INTO "DIMA"."SOFT_IS_INCTALL" (NAME, DATA, VERSION) VALUES ('Discort', TO_DATE('2018-01-01 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '0.0.1')
INSERT INTO "DIMA"."SOFT_IS_INCTALL" (NAME, DATA, VERSION) VALUES ('Moilla', TO_DATE('2017-05-05 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '12.15.2')
INSERT INTO "DIMA"."SOFTWARE" (NAME, DATA, MEMORY) VALUES ('Qt', TO_DATE('2016-12-10 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '2079,36')





[visualization.txt](https://github.com/igortereshchenko/dbis/files/1948895/visualization.txt)

dimon19991 committed:
Add files via upload

 CREATE TABLE "DIMA"."OS_IS_INSTULL" 
   (	"MAC_ADDRESS" VARCHAR2(17 BYTE) NOT NULL ENABLE, 
	"VERSION" VARCHAR2(14 BYTE) NOT NULL ENABLE, 
	 CONSTRAINT "PK_OS_IS_INSTULL" PRIMARY KEY ("MAC_ADDRESS", "VERSION")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS 
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM"  ENABLE, 
	 CONSTRAINT "FK_COMP_HAS_OS" FOREIGN KEY ("MAC_ADDRESS")
	  REFERENCES "DIMA"."СOMPUTER" ("MAC_ADDRESS") ENABLE, 
	 CONSTRAINT "FK_OS_IS_INST" FOREIGN KEY ("VERSION")
	  REFERENCES "DIMA"."OS_WINDOWS" ("VERSION") ENABLE
   ) SEGMENT CREATION IMMEDIATE 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 NOCOMPRESS LOGGING
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM" ;

  
  
    CREATE TABLE "DIMA"."OS_WINDOWS" 
   (	"VERSION" VARCHAR2(14 BYTE) NOT NULL ENABLE, 
	"DESCRIPTION" VARCHAR2(100 BYTE) NOT NULL ENABLE, 
	"LANGUAGE" VARCHAR2(20 BYTE) NOT NULL ENABLE, 
	"SYSTE_BIT_RATE" VARCHAR2(6 BYTE) NOT NULL ENABLE, 
	 CONSTRAINT "PK_OS_WINDOWS" PRIMARY KEY ("VERSION")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS 
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM"  ENABLE
   ) SEGMENT CREATION IMMEDIATE 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 NOCOMPRESS LOGGING
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM" ;
  
  
  
  CREATE TABLE "DIMA"."SOFT_IS_INCTALL" 
   (	"NAME" VARCHAR2(50 BYTE) NOT NULL ENABLE, 
	"DATA" DATE NOT NULL ENABLE, 
	"VERSION" VARCHAR2(14 BYTE) NOT NULL ENABLE, 
	 CONSTRAINT "PK_SOFT_IS_INCTALL" PRIMARY KEY ("NAME", "DATA", "VERSION")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS 
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM"  ENABLE, 
	 CONSTRAINT "FK_SOFT_SOFTWARE" FOREIGN KEY ("NAME", "DATA")
	  REFERENCES "DIMA"."SOFTWARE" ("NAME", "DATA") ENABLE, 
	 CONSTRAINT "FK_SOFT_WINDO" FOREIGN KEY ("VERSION")
	  REFERENCES "DIMA"."OS_WINDOWS" ("VERSION") ENABLE
   ) SEGMENT CREATION IMMEDIATE 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 NOCOMPRESS LOGGING
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM" ;
  
  
    CREATE TABLE "DIMA"."SOFTWARE" 
   (	"NAME" VARCHAR2(50 BYTE) NOT NULL ENABLE, 
	"DATA" DATE NOT NULL ENABLE, 
	"MEMORY" NUMBER(10,2) NOT NULL ENABLE, 
	 CONSTRAINT "PK_SOFTWARE" PRIMARY KEY ("NAME", "DATA")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS 
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM"  ENABLE
   ) SEGMENT CREATION IMMEDIATE 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 NOCOMPRESS LOGGING
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM" ;

  
  
  CREATE TABLE "DIMA"."СOMPUTER" 
   (	"MAC_ADDRESS" VARCHAR2(17 BYTE) NOT NULL ENABLE, 
	"LOGO" VARCHAR2(10 BYTE) NOT NULL ENABLE, 
	"CPU" VARCHAR2(20 BYTE) NOT NULL ENABLE, 
	"MODEL" NUMBER(6,0) NOT NULL ENABLE, 
	 CONSTRAINT "PK_СOMPUTER" PRIMARY KEY ("MAC_ADDRESS")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS 
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM"  ENABLE
   ) SEGMENT CREATION IMMEDIATE 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 NOCOMPRESS LOGGING
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "SYSTEM" ;

  INSERT INTO "DIMA"."СOMPUTER" (MAC_ADDRESS, LOGO, CPU, MODEL) VALUES ('04:7D:7B:97:0C:9F', 'ASUS', 'INTEL I7', '1001')
INSERT INTO "DIMA"."СOMPUTER" (MAC_ADDRESS, LOGO, CPU, MODEL) VALUES ('08:ED:B9:49:B2:E5', 'DELL', 'INTEL I5', '1002')
INSERT INTO "DIMA"."СOMPUTER" (MAC_ADDRESS, LOGO, CPU, MODEL) VALUES ('01:0C:CD:00:00:01', 'SUMSUNG', 'INTEL I3', '1003')
INSERT INTO "DIMA"."СOMPUTER" (MAC_ADDRESS, LOGO, CPU, MODEL) VALUES ('01:0C:CD:01:01:21', 'DELL', 'INTEL I7', '1004')


	INSERT INTO "DIMA"."OS_WINDOWS" (VERSION, DESCRIPTION, LANGUAGE, SYSTE_BIT_RATE) VALUES ('0.0.1', 'GOOD', 'English', '64-bit')
INSERT INTO "DIMA"."OS_WINDOWS" (VERSION, DESCRIPTION, LANGUAGE, SYSTE_BIT_RATE) VALUES ('1.15.2', 'BUG', 'Freanch', '32-bit')
INSERT INTO "DIMA"."OS_WINDOWS" (VERSION, DESCRIPTION, LANGUAGE, SYSTE_BIT_RATE) VALUES ('5.102.84', 'SLOW', 'Englich', '32-bit')
INSERT INTO "DIMA"."OS_WINDOWS" (VERSION, DESCRIPTION, LANGUAGE, SYSTE_BIT_RATE) VALUES ('12.15.2', 'GOOD', 'German', '64-bit')


	INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('04:7D:7B:97:0C:9F', '0.0.1')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('08:ED:B9:49:B2:E5', '5.102.84')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('04:7D:7B:97:0C:9f', '12.15.2')



INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('04:7D:7B:97:0C:9F', '0.0.1')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('08:ED:B9:49:B2:E5', '5.102.84')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('04:7D:7B:97:0C:9F', '12.15.2')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('01:0C:CD:00:00:01', '0.0.1')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('01:0C:CD:00:00:01', '1.15.2')
INSERT INTO "DIMA"."OS_IS_INSTULL" (MAC_ADDRESS, VERSION) VALUES ('01:0C:CD:01:01:21', '12.15.2')


INSERT INTO "DIMA"."SOFTWARE" (NAME, DATA, MEMORY) VALUES ('SKYPE', TO_DATE('2017-12-12 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '705,2')
INSERT INTO "DIMA"."SOFTWARE" (NAME, DATA, MEMORY) VALUES ('Visual Studio', TO_DATE('1017-09-01 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '2028.15')


INSERT INTO "DIMA"."SOFT_IS_INCTALL" (NAME, DATA, VERSION) VALUES ('Skype', TO_DATE('2017-12-12 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '0.0.1')
INSERT INTO "DIMA"."SOFT_IS_INCTALL" (NAME, DATA, VERSION) VALUES ('Team Viver', TO_DATE('2018-10-15 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '12.15.2')
INSERT INTO "DIMA"."SOFT_IS_INCTALL" (NAME, DATA, VERSION) VALUES ('Viber', TO_DATE('2012-10-10 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '1.15.2')
INSERT INTO "DIMA"."SOFT_IS_INCTALL" (NAME, DATA, VERSION) VALUES ('Discort', TO_DATE('2018-01-01 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '0.0.1')
INSERT INTO "DIMA"."SOFT_IS_INCTALL" (NAME, DATA, VERSION) VALUES ('Moilla', TO_DATE('2017-05-05 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '12.15.2')
INSERT INTO "DIMA"."SOFTWARE" (NAME, DATA, MEMORY) VALUES ('Qt', TO_DATE('2016-12-10 00:00:00', 'YYYY-MM-DD HH24:MI:SS'), '2079,36')

dimon19991 committed:
Add files via upload

dimon19991 commented:
create or replace FUNCTION coun_null(vend VENDORS.vend_id%type) 
RETURN INT
AS
    resul INT;
begin
    select 
    count(*) into resul
    from
    VENDORS left join PRODUCTS on
    VENDORS.vend_id = PRODUCTS.vend_id
    left join ORDERITEMS on
    PRODUCTS.PROD_ID = ORDERITEMS.PROD_ID
    where VENDORS.vend_id = vend and order_num is null;

    return resul;
end coun_null;

create or replace PROCEDURE PR_NAME (prod_name in PRODUCTS.PROD_NAME%type, keye out ORDERITEMS.ORDER_NUM%type)
IS
BEGIN
    
     
  select 
    order_num into keye
    from 
    PRODUCTS left join ORDERITEMS on 
    PRODUCTS.prod_id = ORDERITEMS.PROD_ID
    where 
    PRODUCTS.PROD_NAME = prod_name;
END PR_NAME;

dimon19991 committed:
Update laboratory5.sql

dimon19991 committed:
Update laboratory4.sql
